test/DynamoCoreTests/Graph/Workspaces/HomeWorkspaceModelTests.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System.IO;
+using Dynamo.Graph.Workspaces;
+using Dynamo.Models;
+using NUnit.Framework;
+
+namespace Dynamo.Tests
+{
+    /// <summary>
+    /// Class containing tests for the HomeWorkspaceModel class
+    /// </summary>
+    [TestFixture]
+    class HomeWorkspaceModelTests : DynamoModelTestBase
+    {
+        public void OpenTestFile(string folder, string fileName)
+        {
+            var examplePath = Path.Combine(TestDirectory, folder, fileName);
+            OpenModel(examplePath);
+        }
+
+        [Test]
+        [Category("UnitTests")]
+        public void OnPreviewGraphCompletedTest()
+        {
+            //Loads a file with a valid graph on the Home workspace
+            OpenTestFile(@"core\number", "TestBigNumber.dyn");
+            var homeWorkspace = CurrentDynamoModel.CurrentWorkspace as HomeWorkspaceModel;
+            Assert.NotNull(homeWorkspace);
+
+            //Gets results from raised event
+            bool eventWasRaised = false;
+            string objectFileName = "";
+            homeWorkspace.SetNodeDeltaState += delegate(object workspace, DeltaComputeStateEventArgs d)
```

Comment:
When do we unsubscribe?

Reply:
At the moment we don't, I'll look into it and update the pr